src/tir/schedule/analysis/analysis.cc

```diff
@@ -363,6 +363,26 @@ void CheckCompleteOrReductionBlock(const ScheduleState& self, const StmtSRef& bl
                                          reduction_block_error_code);
 }
 
+StmtSRef GetInnermostDominantSubBlockSRef(const ScheduleState& self, const StmtSRef& block_sref) {
+  // return the innermost dominant sub-block.
+  const BlockNode* block = TVM_SREF_TO_BLOCK(block, block_sref);
+  bool all_iter_vars_data_parallel = true;
+  for (const IterVar& iter_var : block->iter_vars) {
+    if (iter_var->iter_type != kDataPar) {
+      all_iter_vars_data_parallel = false;
+    }
+  }
+  Array<StmtSRef> child_block_srefs = GetChildBlockSRefOnSRefTree(self, block_sref);
+  if (!block->init.defined() && child_block_srefs.size() == 1 && all_iter_vars_data_parallel) {
+    const StmtSRef& child_block_sref = child_block_srefs[0];
+    if (IsDominantBlock(self->GetBlockScope(block_sref), child_block_sref)) {
```

Comment:
This is to avoid something like:
```
with T.block("outer"):
    vi, vj = T.axis.remap("SS", [i, j])
    b[vi, vj, 0] = b[vi, vj, 1] + b[vi, vj, 2]
    for k, l in T.grid(16, 16):
        with T.block("inner"):
            vk, vl = T.axis.remap("SR", [k, l])
            with T.init():
                b[vi, vj, vk] = 0.0
            b[vi, vj, vk] = b[vi, vj, vk] + a[vi, vj, vk, vl]
```

But unfortunately the `IsDominantBlock` return true here...

Reply:
Okay... To avoid this stuff, `IsDominantBlock` doesn't work for sure, since `IsDominantBlock("inner")` checks “whether `inner` is the only block writing to `b` under `outer`,” while `b[vi, vj, 0] = b[vi, vj, 1] + b[vi, vj, 2]` isn't wrapped by any sub-block.

IMO an alternative is to require `outer` to have single child on the AST. What do you think of this idea?

Reply:
Yes, I checked the implementation of `IsDominantBlock` I found this case was not considered. But I wonder if it's desired behavior?

Reply:
Checking AST sounds good and I'm working on that.

Reply:
> Yes, I checked the implementation of `IsDominantBlock` I found this case was not considered. But I wonder if it's desired behavior?

Right. “Block `B` is dominant” here means `B` is the only _writer block_ of all the buffer it writes into, under _the scope of `B`’s parent block_. Hence we check all blocks under the parent block of `B`.

In case some `BufferStore` is not wrapped by a sub-block, the check indeed misses that `BufferStore`... I guess we expect all such `BufferStore` to be wrapped by some block, and that might explain why we only check the blocks.

Reply:
We can skip this corner case and use `IsDominantBlock` for now. Since this case is a known problem in TIR and I will fix it together.